code/modules/hippie/recycling/trapdoor.dm

```diff
@@ -0,0 +1,171 @@
+//Trap Door
+/obj/machinery/disposal/trapdoor
+	name = "trapdoor"
+	desc = "Almost like a door, but on floor."
+	density = FALSE
+	icon = 'icons/hippie/obj/atmospherics/trapdoor.dmi'
+	icon_state = "closed"
+	mode = FALSE
+	layer = 2.01
+	var/id = 1
+	var/auto_close = 1200
+	var/auto_close_on_mob = 100
+	var/sound_open = 'sound/hippie/machines/blast_door.ogg'
+	var/sound_close = 'sound/hippie/machines/blast_door.ogg'
+	var/open = FALSE
+
+/obj/machinery/disposal/trapdoor/proc/open()
+	if(flushing)
+		return
+	if(open)
+		return
+	open = TRUE
+	flick("opening", src)
+	playsound(loc, sound_open, 100, 1)
+	icon_state = "open"
+	spawn(5)
+		for(var/mob/living/M in loc)
+			if(!M.floating)
+				M.forceMove(src)
+				trap_flush()
+				if(auto_close_on_mob)
+					spawn(auto_close_on_mob)
+						close()
+		for(var/obj/item/O in loc)
+			if(!O.throwing || !O.anchored)
+				O.forceMove(src)
+				trap_flush()
+	if(auto_close)
+		spawn(auto_close)
+			close()
+	return 1
+
+
+/obj/machinery/disposal/trapdoor/proc/close()
+	if(open)
+		flick("closing", src)
+		icon_state = "closed"
+		playsound(loc, sound_open, 100, 1)
+		open = FALSE
+		return 1
+
+/obj/machinery/disposal/trapdoor/New(loc,var/obj/structure/disposalconstruct/make_from)
+	..()
+	stored.ptype = DISP_END_CHUTE
+	spawn(5)
+		trunk = locate() in loc
+		if(trunk)
+			trunk.linked = src
+
+/obj/machinery/disposal/trapdoor/Crossed(AM as mob|obj)
+	if(open)
+		if(istype(AM, /mob/living))
+			var/mob/living/M = AM
+			if(M.floating)
+				return
+			M.forceMove(src)
+			trap_flush()
+			if(auto_close_on_mob)
+				spawn(auto_close_on_mob)
+					close()
+			return
+		if(istype(AM, /obj/item))
+			var/obj/item/O = AM
+			spawn(5)
+				if(O.throwing || O.anchored)
+					return
+				else if(O.loc == src.loc)
+					O.forceMove(src)
+					trap_flush()
+
+/obj/machinery/disposal/trapdoor/MouseDrop_T(mob/living/target, mob/living/user)
+	if (!open)
+		return
+	if(istype(target))
+		push_mob_in(target, user)
+		return 1
+
+/obj/machinery/disposal/trapdoor/proc/push_mob_in(mob/living/target, mob/living/carbon/human/user)
+	if(target.buckled)
+		return
+	add_fingerprint(user)
+	if(user == target)
+		if(target.floating)
+			user.visible_message("[user] is attempting to dive into [name].", \
+				"<span class='notice'>You start diving into [name]...</span>")
+			if(!do_mob(target, user, 10))
+				return
+			target.forceMove(src)
+			user.visible_message("[user] dives into [name].", \
+				"<span class='notice'>You dive into [name].</span>")
+			sleep(5)
```

Comment:
Please use the `addtimer` and `callback` system to do this instead of using spawns, you can find more information about this [here](https://github.com/HippieStation/HippieStation13/blob/master/code/datums/callback.dm)